### PR TITLE
Fix #1092

### LINKED ISSUE #1092: cyclic dependencies
if we have an object like : 
         obj  =      {
			a : ['c',a],
			b : ['a',b],
			c : ['b',c]
		},
where a,b,c are names of functions, shouldn't this throw cyclic depenedency error if passed to async.auto ?

### PR DESCRIPTION
**Pull request by b-paul**
I believe this does the job requested.

The error text does not report which key(s) are involved in the cycle if one is detected. I haven't spent any thought on that, but will try to figure it out if needed.

### CODE REVIEW

**Comment by megawac** [comment]
Looks good, /cc @aearly

**Comment by megawac** [comment]
Also as we've started using some external dependencies in v2, do you think it makes sense to use an existing dag implementation in auto?

File: lib/auto.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import arrayEach from 'lodash/_arrayEach';
 import forOwn from 'lodash/forOwn';
-import indexOf from 'lodash/_baseIndexOf';
+import indexOf from 'lodash/indexOf';
```
Comment (aearly) [inline lib/auto.js]: Why did you change this?

**Comment by aearly** [comment]
LGTM2.

If there's a tiny DAG library out there that's easy to integrate, maybe.  This code has the advantage of already working with what we've defined to be a "graph", rather than having to construct a graph to verify that it's acyclic.

File: lib/auto.js (hunk 1 above)
Comment (b-paul) [inline lib/auto.js]: @aearly I wrote my `checkForDeadlocks` assuming I had the lodash/JavaScript standard semantics. Got test failures because I hadn't included start index.

Thought I was better off importing the public API method, in case someone else were to come and repeat my mistake at another time

Comment (megawac) [inline lib/auto.js]: We weren't importing indexof to avoid including the lodash internal
toInteger implementation

**Comment by b-paul** [comment]
@megawac @aearly I see your concerns about `indexOf`. Thanks for the explanation; I should have guessed it was that way for a reason. I've pushed a commit that reverts the change and fixes the invocation to match

**Comment by megawac** [comment]
Good stuff! thanks
